### PR TITLE
Avoid duplicating Alchemy.publish_targets in development environment

### PR DESCRIPTION
**Pull request by dbwinger**

### CODE REVIEW

File: lib/alchemy/pg_search/engine.rb [hunk 1]
```diff
@@ -14,6 +14,8 @@ class Engine < ::Rails::Engine
         # We need to have the search methods present in all Alchemy controllers
         Alchemy::BaseController.send(:include, Alchemy::PgSearch::ControllerMethods)
 
+        # In development environment, this runs on every code reload, so avoid multiple reindexing jobs
+        Alchemy.publish_targets.delete_if { |t| t.name == 'Alchemy::PgSearch::IndexPageJob' }
         # reindex the page after it was published
         Alchemy.publish_targets << Alchemy::PgSearch::IndexPageJob
```
Comment (tvdeyen) [inline lib/alchemy/pg_search/engine.rb]: If would prefer a check if the target is present

```suggestion
        # In development environment, this runs on every code reload, so avoid multiple reindexing jobs
        unless Alchemy.publish_targets.map(&:name).include? 'Alchemy::PgSearch::IndexPageJob'
          # reindex the page after it was published
          Alchemy.publish_targets << Alchemy::PgSearch::IndexPageJob
        end
```

**Review by tvdeyen — COMMENTED** [review]
Thanks. For now this should be fine (with a little refactoring), but ideally we would fix this in Alchemy and use a `Set` for the publish targets.

**Comment by dbwinger** [comment]
> Thanks. For now this should be fine (with a little refactoring), but ideally we would fix this in Alchemy and use a `Set` for the publish targets.

FYI, `Alchemy.publish_targets` already is using `Set` but apparently it's not considering the multiple additions as duplicates.

**Review by tvdeyen — APPROVED** [review]

**Review by tvdeyen — APPROVED** [review]

**Comment by tvdeyen** [comment]
@dbwinger released as 5.2

**Pull request merged**